### ISSUE TITLE
nox tests fail

### ISSUE THREAD

**Issue by SurferJeffAtGoogle** [googleapis/synthtool#338]
Python 3.7
Windows

Full test log:
```
PS C:\Users\rennie\gitrepos\synthtool> nox
nox > Running session blacken
nox > Creating virtual environment (virtualenv) using python.exe in .nox\blacken
nox > pip install black
nox > black synthtool tests
All done! ✨ 🍰 ✨
33 files left unchanged.
nox > Session blacken was successful.
nox > Running session lint
nox > Creating virtual environment (virtualenv) using python.exe in .nox\lint
nox > pip install mypy flake8 black
nox > pip install -e .
Obtaining file:///C:/Users/rennie/gitrepos/synthtool
Requirement already satisfied: click in c:\users\rennie\gitrepos\synthtool\.nox\lint\lib\site-packages (from gcp-synthtool==2019.10.17) (7.0)
Collecting colorlog
  Using cached https://files.pythonhosted.org/packages/68/4d/892728b0c14547224f0ac40884e722a3d00cb54e7a146aea0b3186806c9e/colorlog-4.0.2-py2.py3-none-any.whl
Collecting jinja2
  Downloading https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl (125kB)
     |████████████████████████████████| 133kB 6.8MB/s
Collecting packaging
  Using cached https://files.pythonhosted.org/packages/cf/94/9672c2d4b126e74c4496c6b3c58a8b51d6419267be9e70660ba23374c875/packaging-19.2-py2.py3-none-any.whl
Collecting PyYAML
  Downloading https://files.pythonhosted.org/packages/58/1f/00682a829890131bcc71755e78d595e439bd6b291ea9bd5ced93ed57cc3d/PyYAML-5.2-cp36-cp36m-win_amd64.whl (214kB)
     |████████████████████████████████| 215kB 6.4MB/s
Collecting requests
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting protobuf
  Downloading https://files.pythonhosted.org/packages/78/c5/5f64362524497c0ab5f532f58d605c9a6f75a7313974893f6c3168903dfe/protobuf-3.11.1-cp36-cp36m-win_amd64.whl (1.1MB)
     |████████████████████████████████| 1.1MB 6.8MB/s
Collecting colorama; sys_platform == "win32"
  Using cached https://files.pythonhosted.org/packages/4f/a6/728666f39bfff1719fc94c481890b2106837da9318031f71a8424b662e12/colorama-0.4.1-py2.py3-none-any.whl
Collecting MarkupSafe>=0.23
  Downloading https://files.pythonhosted.org/packages/b9/82/833c7714951bff8f502ed054e6fbd8bd00e083d1fd96de6a46905cf23378/MarkupSafe-1.1.1-cp36-cp36m-win_amd64.whl
Collecting pyparsing>=2.0.2
  Downloading https://files.pythonhosted.org/packages/c0/0c/fc2e007d9a992d997f04a80125b0f183da7fb554f1de701bbb70a8e7d479/pyparsing-2.4.5-py2.py3-none-any.whl (67kB)
     |████████████████████████████████| 71kB 2.3MB/s
Collecting six
  Downloading https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Downloading https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl (125kB)
     |████████████████████████████████| 133kB 6.4MB/s
Collecting certifi>=2017.4.17
  Downloading https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl (156kB)
     |████████████████████████████████| 163kB 1.3MB/s
Requirement already satisfied: setuptools in c:\users\rennie\gitrepos\synthtool\.nox\lint\lib\site-packages (from protobuf->gcp-synthtool==2019.10.17) (42.0.2)
Installing collected packages: colorama, colorlog, MarkupSafe, jinja2, pyparsing, six, packaging, PyYAML, chardet, idna, urllib3, certifi, requests, protobuf, gcp-synthtool
  Running setup.py develop for gcp-synthtool
Successfully installed MarkupSafe-1.1.1 PyYAML-5.2 certifi-2019.11.28 chardet-3.0.4 colorama-0.4.1 colorlog-4.0.2 gcp-synthtool idna-2.8 jinja2-2.10.3 packaging-19.2 protobuf-3.11.1 pyparsing-2.4.5 requests-2.22.0 six-1.13.0 urllib3-1.25.7
nox > black --check synthtool tests
All done! ✨ 🍰 ✨
33 files would be left unchanged.
nox > flake8 synthtool tests
nox > mypy synthtool
synthtool\gcp\gapic_microgenerator.py:145: error: Module has no attribute "getuid"; maybe "getpid" or "getppid"?
Found 1 error in 1 file (checked 26 source files)
nox > Command mypy synthtool failed with exit code 1
nox > Session lint failed.
nox > Running session test
nox > Creating virtual environment (virtualenv) using python.exe in .nox\test
nox > pip install pytest pytest-cov
nox > pip install -e .
Obtaining file:///C:/Users/rennie/gitrepos/synthtool
Collecting click
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting colorlog
  Using cached https://files.pythonhosted.org/packages/68/4d/892728b0c14547224f0ac40884e722a3d00cb54e7a146aea0b3186806c9e/colorlog-4.0.2-py2.py3-none-any.whl
Collecting jinja2
  Using cached https://files.pythonhosted.org/packages/65/e0/eb35e762802015cab1ccee04e8a277b03f1d8e53da3ec3106882ec42558b/Jinja2-2.10.3-py2.py3-none-any.whl
Requirement already satisfied: packaging in c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages (from gcp-synthtool==2019.10.17) (19.2)
Collecting PyYAML
  Using cached https://files.pythonhosted.org/packages/58/1f/00682a829890131bcc71755e78d595e439bd6b291ea9bd5ced93ed57cc3d/PyYAML-5.2-cp36-cp36m-win_amd64.whl
Collecting requests
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting protobuf
  Using cached https://files.pythonhosted.org/packages/78/c5/5f64362524497c0ab5f532f58d605c9a6f75a7313974893f6c3168903dfe/protobuf-3.11.1-cp36-cp36m-win_amd64.whl
Requirement already satisfied: colorama; sys_platform == "win32" in c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages (from colorlog->gcp-synthtool==2019.10.17) (0.4.1)
Collecting MarkupSafe>=0.23
  Using cached https://files.pythonhosted.org/packages/b9/82/833c7714951bff8f502ed054e6fbd8bd00e083d1fd96de6a46905cf23378/MarkupSafe-1.1.1-cp36-cp36m-win_amd64.whl
Requirement already satisfied: six in c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages (from packaging->gcp-synthtool==2019.10.17) (1.13.0)
Requirement already satisfied: pyparsing>=2.0.2 in c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages (from packaging->gcp-synthtool==2019.10.17) (2.4.5)
Collecting chardet<3.1.0,>=3.0.2
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting certifi>=2017.4.17
  Using cached https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Using cached https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl
Requirement already satisfied: setuptools in c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages (from protobuf->gcp-synthtool==2019.10.17) (42.0.2)
Installing collected packages: click, colorlog, MarkupSafe, jinja2, PyYAML, chardet, certifi, idna, urllib3, requests, protobuf, gcp-synthtool
  Running setup.py develop for gcp-synthtool
Successfully installed MarkupSafe-1.1.1 PyYAML-5.2 certifi-2019.11.28 chardet-3.0.4 click-7.0 colorlog-4.0.2 gcp-synthtool idna-2.8 jinja2-2.10.3 protobuf-3.11.1 requests-2.22.0 urllib3-1.25.7
nox > pytest --cov-report term-missing --cov synthtool tests
================================================== test session starts ==================================================
platform win32 -- Python 3.6.4, pytest-5.3.1, py-1.8.0, pluggy-0.13.1
rootdir: C:\Users\rennie\gitrepos\synthtool
plugins: cov-2.8.1
collected 40 items                                                                                                       

tests\test_common.py ...                                                                                           [  7%]
tests\test_git.py ........                                                                                         [ 27%]
tests\test_metadata.py .....                                                                                       [ 40%]
tests\test_metadata_protos.py ..                                                                                   [ 45%]
tests\test_templates.py ..F.......                                                                                 [ 70%]
tests\test_tracked_paths.py .                                                                                      [ 72%]
tests\test_transforms.py ...FFFFFFFF                                                                               [100%]

======================================================= FAILURES ========================================================
_______________________________________________ test_render_preserve_mode _______________________________________________

    def test_render_preserve_mode():
        """
        Test that rendering templates correctly preserve file modes.
        """
        source_file = FIXTURES / "executable.j2"
        source_mode = source_file.stat().st_mode
    
        # Verify source fixture has execute permission for USER
>       assert source_mode & stat.S_IXUSR
E       assert (33206 & 64)
E        +  where 64 = stat.S_IXUSR

tests\test_templates.py:52: AssertionError
_________________________________________ test__expand_paths[**/*.py-expected3] _________________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__expand_paths______py_exp0')
input = '**/*.py', expected = ['b.py', 'dira/f.py', 'dirb/suba/g.py']

    @pytest.mark.parametrize(
        ["input", "expected"],
        [
            ("a.txt", ["a.txt"]),
            ("*", ["a.txt", "b.py", "c.md", "dira", "dirb"]),
            ("*.py", ["b.py"]),
            ("**/*.py", ["b.py", "dira/f.py", "dirb/suba/g.py"]),
            (
                "**/*",
                [
                    "a.txt",
                    "b.py",
                    "c.md",
                    "dira",
                    "dira/e.txt",
                    "dira/f.py",
                    "dirb",
                    "dirb/suba",
                    "dirb/suba/g.py",
                ],
            ),
        ],
    )
    def test__expand_paths(expand_path_fixtures, input, expected):
        paths = sorted([str(x) for x in transforms._expand_paths(input)])
>       assert paths == expected
E       AssertionError: assert ['b.py', 'dir...\\suba\\g.py'] == ['b.py', 'dir...rb/suba/g.py']
E         At index 1 diff: 'dira\\f.py' != 'dira/f.py'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:79: AssertionError
__________________________________________ test__expand_paths[**/*-expected4] ___________________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__expand_paths______expect0')
input = '**/*', expected = ['a.txt', 'b.py', 'c.md', 'dira', 'dira/e.txt', 'dira/f.py', ...]

    @pytest.mark.parametrize(
        ["input", "expected"],
        [
            ("a.txt", ["a.txt"]),
            ("*", ["a.txt", "b.py", "c.md", "dira", "dirb"]),
            ("*.py", ["b.py"]),
            ("**/*.py", ["b.py", "dira/f.py", "dirb/suba/g.py"]),
            (
                "**/*",
                [
                    "a.txt",
                    "b.py",
                    "c.md",
                    "dira",
                    "dira/e.txt",
                    "dira/f.py",
                    "dirb",
                    "dirb/suba",
                    "dirb/suba/g.py",
                ],
            ),
        ],
    )
    def test__expand_paths(expand_path_fixtures, input, expected):
        paths = sorted([str(x) for x in transforms._expand_paths(input)])
>       assert paths == expected
E       AssertionError: assert ['a.txt', 'b....a\\f.py', ...] == ['a.txt', 'b....ra/f.py', ...]
E         At index 4 diff: 'dira\\e.txt' != 'dira/e.txt'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:79: AssertionError
_____________________________________ test__expand_paths_with_root[e.txt-expected0] _____________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__expand_paths_with_root_e0')
input = 'e.txt', expected = ['dira/e.txt']

    @pytest.mark.parametrize(
        ["input", "expected"],
        [("e.txt", ["dira/e.txt"]), ("*", ["dira/e.txt", "dira/f.py"])],
    )
    def test__expand_paths_with_root(expand_path_fixtures, input, expected):
        paths = sorted([str(x) for x in transforms._expand_paths(input, root="dira")])
>       assert paths == expected
E       AssertionError: assert ['dira\\e.txt'] == ['dira/e.txt']
E         At index 0 diff: 'dira\\e.txt' != 'dira/e.txt'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:88: AssertionError
_______________________________________ test__expand_paths_with_root[*-expected1] _______________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__expand_paths_with_root__0')
input = '*', expected = ['dira/e.txt', 'dira/f.py']

    @pytest.mark.parametrize(
        ["input", "expected"],
        [("e.txt", ["dira/e.txt"]), ("*", ["dira/e.txt", "dira/f.py"])],
    )
    def test__expand_paths_with_root(expand_path_fixtures, input, expected):
        paths = sorted([str(x) for x in transforms._expand_paths(input, root="dira")])
>       assert paths == expected
E       AssertionError: assert ['dira\\e.txt', 'dira\\f.py'] == ['dira/e.txt', 'dira/f.py']
E         At index 0 diff: 'dira\\e.txt' != 'dira/e.txt'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:88: AssertionError
__________________________________________________ test__filter_files ___________________________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__filter_files0')

    def test__filter_files(expand_path_fixtures):
        files = sorted(
            [str(x) for x in transforms._filter_files(transforms._expand_paths("**/*"))]
        )
    
>       assert files == [
            "a.txt",
            "b.py",
            "c.md",
            "dira/e.txt",
            "dira/f.py",
            "dirb/suba/g.py",
        ]
E       AssertionError: assert ['a.txt', 'b....\\suba\\g.py'] == ['a.txt', 'b....rb/suba/g.py']
E         At index 3 diff: 'dira\\e.txt' != 'dira/e.txt'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:96: AssertionError
_________________________________________________ test__file_copy_mode __________________________________________________

executable_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__file_copy_mode0')

    def test__file_copy_mode(executable_fixtures):
        executable = "exec.sh"
        destination = executable_fixtures / "exec_copy.sh"
    
        transforms.move([executable], destination)
    
        # Check if destination file has execute permission for USER
>       assert destination.stat().mode & stat.S_IXUSR
E       AssertionError: assert (33206 & 64)
E        +  where 33206 = <py._path.local.Stat object at 0x00000255EDC75F28>.mode
E        +    where <py._path.local.Stat object at 0x00000255EDC75F28> = <bound method LocalPath.stat of local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__file_copy_mode0\\exec_copy.sh')>()
E        +      where <bound method LocalPath.stat of local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__file_copy_mode0\\exec_copy.sh')> = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__file_copy_mode0\\exec_copy.sh').stat
E        +  and   64 = stat.S_IXUSR

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:113: AssertionError
__________________________________________________ test__move_to_dest ___________________________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__move_to_dest0')

    def test__move_to_dest(expand_path_fixtures):
        tmp_path = Path(str(expand_path_fixtures))
        _tracked_paths.add(expand_path_fixtures)
        dest = Path(str(expand_path_fixtures / "dest"))
    
        transforms.move(tmp_path, dest, excludes=["dira/f.py"])
    
        files = sorted([str(x) for x in transforms._expand_paths("**/*", root="dest")])
    
        # Assert destination does not contain dira/f.py (excluded)
>       assert files == [
            "dest/a.txt",
            "dest/b.py",
            "dest/c.md",
            "dest/dira",
            "dest/dira/e.txt",
            "dest/dirb",
            "dest/dirb/suba",
            "dest/dirb/suba/g.py",
        ]
E       AssertionError: assert ['dest\\a.txt...t\\dirb', ...] == ['dest/a.txt'...st/dirb', ...]
E         At index 0 diff: 'dest\\a.txt' != 'dest/a.txt'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:126: AssertionError
_______________________________________________ test__move_to_dest_subdir _______________________________________________

expand_path_fixtures = local('C:\\Users\\rennie\\AppData\\Local\\Temp\\pytest-of-rennie\\pytest-0\\test__move_to_dest_subdir0')

    def test__move_to_dest_subdir(expand_path_fixtures):
        tmp_path = Path(str(expand_path_fixtures))
        _tracked_paths.add(expand_path_fixtures)
        dest = Path(str(expand_path_fixtures / "dest/dira"))
    
        # Move to a different dir to make sure that move doesn't depend on the cwd
        os.chdir(tempfile.gettempdir())
        transforms.move(tmp_path / "dira", dest, excludes=["f.py"])
    
        os.chdir(str(tmp_path))
        files = sorted([str(x) for x in transforms._expand_paths("**/*", root="dest")])
    
        # Assert destination does not contain dira/f.py (excluded)
>       assert files == ["dest/dira", "dest/dira/e.txt"]
E       AssertionError: assert ['dest\\dira'...\dira\\e.txt'] == ['dest/dira',...t/dira/e.txt']
E         At index 0 diff: 'dest\\dira' != 'dest/dira'
E         Use -v to get the full diff

C:\Users\rennie\gitrepos\synthtool\tests\test_transforms.py:151: AssertionError

----------- coverage: platform win32, python 3.6.4-final-0 -----------
Name                                    Stmts   Miss Branch BrPart  Cover   Missing
-----------------------------------------------------------------------------------
synthtool\__init__.py                      11      1      2      1    85%   31, 30->31
synthtool\__main__.py                      35     16      6      1    49%   29-30, 55, 64-83, 87, 86->87
synthtool\_tracked_paths.py                14      3      4      1    78%   37-39, 34->39
synthtool\cache.py                          5      3      0      0    40%   19-21
synthtool\gcp\__init__.py                   9      0      0      0   100%
synthtool\gcp\artman.py                    57     37     14      0    28%   32-36, 40-44, 51-55, 59, 86-143, 146-156, 161-162, 167
synthtool\gcp\common.py                   111     33     40      6    65%   41-54, 57, 61-63, 68-75, 78, 82-84, 87, 93-99, 121, 192, 112->exit, 116->115, 120->121, 187->191, 188->187, 191->192
synthtool\gcp\discogapic_generator.py      47     31     14      0    26%   29, 38, 41, 46, 49, 52, 58-115, 118-119
synthtool\gcp\gapic_generator.py          153    124     62      0    13%   39-41, 50, 53, 58, 61, 64, 79-204, 207-218, 221-234, 276-361
synthtool\gcp\gapic_microgenerator.py     100     77     44      0    16%   42-46, 55, 58, 61, 64, 80-203, 206-217, 220-233, 236-246
synthtool\languages\__init__.py             0      0      0      0   100%
synthtool\languages\java.py                53     53      6      0     0%   15-169
synthtool\languages\node.py                15     10      3      0    28%   28-42, 58
synthtool\languages\python.py               9      9      0      0     0%   15-50
synthtool\languages\ruby.py                25     25      4      0     0%   15-73
synthtool\log.py                           40      9      2      1    76%   19-20, 31-32, 44, 52, 56, 64, 68, 76->92
synthtool\metadata.py                      27      1      0      0    96%   70
synthtool\protos\__init__.py                0      0      0      0   100%
synthtool\protos\metadata_pb2.py           60      0      2      0   100%
synthtool\shell.py                         11      8      4      0    20%   21-39
synthtool\sources\__init__.py               0      0      0      0   100%
synthtool\sources\git.py                   55     18     18      1    63%   54-86, 103, 102->103
synthtool\sources\templates.py             76     11     22      4    79%   85, 100, 103-113, 134, 81->85, 99->100, 101->103, 132->134
synthtool\tmp.py                           16      3      2      0    72%   34-36
synthtool\transforms.py                    99     37     56      8    63%   33, 54, 80-91, 109, 127, 150, 167, 173, 182-195, 202-216, 32->33, 49->54, 108->109, 126->127, 149->150, 164->148, 166->167, 172->173
synthtool\update_check.py                  29      6      8      4    73%   34-36, 45, 50, 61, 33->34, 41->45, 49->50, 58->61
-----------------------------------------------------------------------------------
TOTAL                                    1057    515    313     27    46%

============================================= 9 failed, 31 passed in 3.60s ==============================================
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\rennie\AppData\Local\Programs\Python\Python36\Lib\logging\__init__.py", line 994, in emit
    stream.write(msg)
  File "c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages\colorama\ansitowin32.py", line 41, in write
    self.__convertor.write(text)
  File "c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages\colorama\ansitowin32.py", line 162, in write
    self.write_and_convert(text)
  File "c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages\colorama\ansitowin32.py", line 187, in write_and_convert
    self.write_plain_text(text, cursor, start)
  File "c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages\colorama\ansitowin32.py", line 195, in write_plain_text
    self.wrapped.write(text[start:end])
  File "c:\users\rennie\gitrepos\synthtool\.nox\test\lib\site-packages\_pytest\capture.py", line 427, in write
    self.buffer.write(obj)
ValueError: I/O operation on closed file
Call stack:
  File "c:\users\rennie\gitrepos\synthtool\synthtool\tmp.py", line 36, in cleanup
    log.debug(f"Cleaned up {len(_tempdirs)} temporary directories.")
  File "c:\users\rennie\gitrepos\synthtool\synthtool\log.py", line 48, in debug
    logger.debug(*args, **kwargs)
Message: 'Cleaned up 8 temporary directories.'
Arguments: ()
nox > Command pytest --cov-report term-missing --cov synthtool tests failed with exit code 1
nox > Session test failed.
nox > Running session generate_protos
nox > Creating virtual environment (virtualenv) using python.exe in .nox\generate_protos
nox > pip install grpcio-tools
nox > python -m grpc_tools.protoc -Isynthtool/protos --python_out=synthtool/protos synthtool/protos/metadata.proto
nox > Session generate_protos was successful.
nox > Ran multiple sessions:
nox > * blacken: success
nox > * lint: failed
nox > * test: failed
nox > * generate_protos: success
PS C:\Users\rennie\gitrepos\synthtool>
```

**Comment by bcoe** [comment]
@SurferJeffAtGoogle usually I'd call test failures a P1, because it can block further work on the library, but could imagine calling this a P2 since it's our own library (and we're near holidays and the 7 day response might be hard).

**Comment by busunkim96** [comment]
Synthtool assumes paths are Unix-style, which seems to be causing most or all of the test failures above.

**Comment by JustinBeckwith** [comment]
Good to see a fix in #339, but a little concerned that this happened.
- We should have Windows tests as part of our shared kokoro config for each language
- That config should be shared across all python libraries (like it is for node) via synthtool (lol)

The fact that we didn't have Windows testing just kinda hanging around gives me the willies.  Are the test scripts not shared via synthtool templates?  Why aren't we testing Windows by default here?
